pandas/core/indexes/api.py

```diff
@@ -213,14 +215,41 @@ def conv(i):
 
     if kind == "special":
         result = indexes[0]
+        first = result
+
+        dtis = [x for x in indexes if isinstance(x, DatetimeIndex)]
```

Comment:
this is super special casey. umm

Reply:
yah its kinda gross but the best i could figure out.  it gets a bit better once a deprecation is enforced in 2.0, and we could consider changing the raising case

Reply:
ok will look in a bit more detail